rcl/include/rcl/wait.h

```diff
@@ -266,6 +272,10 @@ rcl_wait_set_clear(rcl_wait_set_t * wait_set);
  *
  * This can be called on an uninitialized (zero initialized) wait set.
  *
+ * If RCL_RET_BAD_ALLOC is returned, i.e. allocation failed, the wait_set
+ * may be *partially* allocated and therefore, it must still be cleaned up
+ * with _fini.
```

Comment:
This is just documenting existing behavior. I noticed this when examining the existing allocation macros -- when they fail, they just set the *field currently being allocated* to NULL and then return RET_BAD_ALLOC. All existing fields of the structure are left as they are and thus need to be deallocated as usual.

I agree with you that this is sub-optimal behavior, but it was there before, and ist not trivial to handle otherwise, so I don't think I can change this right now but at least I wanted to document it.